src/AnimatedSprite.cpp

```diff
@@ -81,24 +87,50 @@ sf::FloatRect AnimatedSprite::getGlobalBounds() const
 
 void AnimatedSprite::setHighlight(bool h)
 {
-    if (h)
-    {
-        _highlight.setPosition(getPosition());
-        auto b = getGlobalBounds();
-        _highlight.setSize(sf::Vector2f{ b.width, b.height });
-    }
-    else
+    _isHighlighted = h;
+    setHighlightPosition();
+}
+
+void AnimatedSprite::draw(  sf::RenderTarget& target, 
+                            sf::RenderStates states     ) const
+{
+
+    states.transform *= getTransform();
+    target.draw(_sprite, states);
+
+    if(_isHighlighted)
```

Comment:
Yeah, this was the premature optimization asshole inside of me. Neither the boolean nor the boolean-check are needed if you set the highlight's size to zero when you want to turn it off.

Reply:
Setting the highlight size to 0 breaks the bounds checking I am trying to use this for.